src/environments/array.js

```diff
@@ -46,6 +56,13 @@ function parseArray(
     let row = [];
     const body = [row];
     const rowGaps = [];
+    const numHLinesBeforeRow = [0];
+
+    // Test for \hline at the top of the array.
+    while (nextTokenIsHLine(parser)) {
+        numHLinesBeforeRow[0] += 1;
+    }
```

Comment:
```js
    const numHLinesBeforeRow = [];

    numHLinesBeforeRow.push(getNumHLines(parser));
```
(same for below)?

Reply:
That's an outstanding suggestion. I'll do it.